lib/ansible/plugins/lookup/hashi_vault.py

```diff
@@ -271,6 +300,54 @@ def auth_approle(self, **kwargs):
 
         self.client.auth_approle(role_id, secret_id)
 
+    def auth_aws_iam(self, **kwargs):
+        access_key = kwargs.get('access_key')
+        secret_key = kwargs.get('secret_key')
+        session_token = kwargs.get('session_token')
+
+        if access_key is None or secret_key is None:
+            import boto3
+            creds = boto3.Session().get_credentials()
+            if creds is None:
+                raise AnsibleError("Authentication method aws iam requires AWS credentials")
+            access_key, secret_key, session_token = creds.access_key, creds.secret_key, creds.token
+
+        header_value = kwargs.get('header_value')
+
+        mount_point = kwargs.get('mount_point')
+        if mount_point is None:
+            mount_point = 'aws'
+
+        role = kwargs.get('role')
+        if role is None:
+            raise AnsibleError("Authentication method aws iam requires a role")
+
+        region = kwargs.get('region', os.environ.get('AWS_DEFAULT_REGION'))
+        if region is None:
+            region = 'us-east-1'
+
+        self.client.auth_aws_iam(access_key, secret_key, session_token, header_value, mount_point, role, region=region)
+
+    def auth_aws_ec2(self, **kwargs):
+        url = "http://169.254.169.254/latest/dynamic/instance-identity/pkcs7"
+
+        try:
+            pkcs7 = requests.get(url, timeout=0.1).text.replace('\n', '')
+        except requests.exceptions.ConnectionError:
+            raise AnsibleError("hashi_vault lookup plugin failed to connect to http://169.254.169.254, Make sure you are running from a AWS EC2 instance")
+
+        nonce = kwargs.get('nonce')
+
+        role = kwargs.get('role')
+        if role is None:
+            raise AnsibleError("Authentication method aws ec2 requires a role")
+
+        mount_point = kwargs.get('mount_point')
+        if mount_point is None:
+            mount_point = 'aws'
+
+        self.client.auth_ec2(pkcs7, nonce, role, mount_point=mount_point)
```

Comment:
From looking at the hvac source code it appears the `client.auth_ec2` method is also deprecated: https://github.com/hvac/hvac/blob/500ce54b8d869a2086396e6cf7c0b6b30cf72f94/hvac/v1/__init__.py#L648

```suggestion
        self.client.auth.aws.ec2_login(pkcs7, nonce, role, use_token=True, mount_point)
```

I haven't tested this, since my setup is not compatible with ec2 logins.